packages/renderer/src/lib/image/PullImage.svelte

```diff
@@ -20,23 +20,34 @@ import RecommendedRegistry from './RecommendedRegistry.svelte';
 const DOCKER_PREFIX = 'docker.io';
 const DOCKER_PREFIX_WITH_SLASH = DOCKER_PREFIX + '/';
 
-let logsPull: Terminal;
-let pullError = '';
-let pullInProgress = false;
-let pullFinished = false;
-let shortnameImages: string[] = [];
-let podmanFQN = '';
-let usePodmanFQN = false;
-let isValidName = true;
+interface Props {
+  imageToPull?: string;
+}
+
+let { imageToPull }: Props = $props();
 
-export let imageToPull: string | undefined = undefined;
+let logsPull: Terminal | undefined = $state();
+let pullError = $state('');
+let pullInProgress = $state(false);
+let pullFinished = $state(false);
+let shortnameImages: string[] = $state([]);
+let podmanFQN = $state('');
+let usePodmanFQN = $state(false);
+let isValidName = $state(true);
 
-$: providerConnections = $providerInfos
-  .map(provider => provider.containerConnections)
-  .flat()
-  .filter(providerContainerConnection => providerContainerConnection.status === 'started');
+let latestTagMessage: string | undefined = $state();
+let imageNameInvalid: string | undefined = $state();
+let imageNameIsInvalid = $state(imageToPull === undefined || imageToPull.trim() === '');
 
-let selectedProviderConnection: ProviderContainerConnectionInfo | undefined;
+// all started connection
+let providerConnections: ProviderContainerConnectionInfo[] = $derived(
+  $providerInfos
```

Comment:
AFAIK I had issues with the `$` usage of stores within state/derived not being reactive.
Did you test like you go the pullImage screen, and you start the podman machine later. is it updating the providerConnections as expected ?

Reply:
Indeed, seems some issues are possible, moving to draft, will try to fix it

Reply:
The problem was coming from the Dropdown, and I applied the fix I found in https://github.com/podman-desktop/podman-desktop/issues/10734#issuecomment-2602654447